### ISSUE TITLE
A warning and two syntax errors

### ISSUE THREAD

**Issue by schoeke** [nerab/dropcaster#6]
pygmy ➜  ~  dropcaster            
DEPRECATION WARNING: this file is deprecated and will be removed. (called from require at /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36)
/home/andrej/.rvm/gems/ruby-2.1.0/gems/dropcaster-0.0.4/lib/dropcaster/channel.rb:156: warning: else without rescue is useless
/usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require': /home/andrej/.rvm/gems/ruby-2.1.0/gems/dropcaster-0.0.4/lib/dropcaster/channel.rb:151: syntax error, unexpected ':', expecting keyword_then or ',' or ';' or '\n' (SyntaxError)
        when nil   : nil
                    ^
/home/andrej/.rvm/gems/ruby-2.1.0/gems/dropcaster-0.0.4/lib/dropcaster/channel.rb:153: syntax error, unexpected keyword_when, expecting keyword_end
        when false : 'No'
            ^
/home/andrej/.rvm/gems/ruby-2.1.0/gems/dropcaster-0.0.4/lib/dropcaster/channel.rb:159: syntax error, unexpected keyword_end, expecting $end
	from /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
	from /home/andrej/.rvm/gems/ruby-2.1.0/gems/dropcaster-0.0.4/lib/dropcaster.rb:15:in `<top (required)>'
	from /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
	from /usr/lib/ruby/1.9.1/rubygems/custom_require.rb:36:in `require'
	from /home/andrej/.rvm/gems/ruby-2.1.0/gems/dropcaster-0.0.4/bin/dropcaster:28:in `<top (required)>'
	from /home/andrej/.rvm/gems/ruby-2.1.0/bin/dropcaster:19:in `load'
	from /home/andrej/.rvm/gems/ruby-2.1.0/bin/dropcaster:19:in `<main>'
	from /home/andrej/.rvm/gems/ruby-2.1.0/bin/ruby_executable_hooks1.9.1:15:in `eval'
	from /home/andrej/.rvm/gems/ruby-2.1.0/bin/ruby_executable_hooks1.9.1:15:in `<main>'

pygmy ➜  ~  gem list

*** LOCAL GEMS ***

activesupport (4.0.2)
atomic (1.1.14)
bundler (1.5.2)
bundler-unload (1.0.2)
dropcaster (0.0.4)
executable-hooks (1.3.1)
gem-wrappers (1.2.4)
i18n (0.6.9)
minitest (4.7.5)
multi_json (1.8.4)
ruby-mp3info (0.8.3)
rubygems-bundler (1.4.2)
rvm (1.11.3.8)
thread_safe (0.1.3)
tzinfo (0.3.38)

**Comment by nerab** [comment]
Is this issue still present? I just tried the latest Ubuntu 14.04 with Ruby 2.2, and it seems to work flawlessly.

If it doesn't for you, could you please run the VM described in this [Vagrantfile](/nerab/dropcaster/blob/master/test/Vagrantfile), and compare it to your environment?

Thanks!

**Comment by schoeke** [comment]
I tested it the other day. Yes, present and kicking on Mac OS X, CentOS and Ubuntu. I will test the vagrant set up and report back.

**Comment by schoeke** [comment]
I tested it and it works in the vagrant box. I checked it on my Ubuntu box. It seems like libxml2-dev was not installed. I installed it, it made no difference for my error initially. But if I test it now with the self-compiled version, I get an error unrelated to dropcaster (Complains about a VBR header).

Testing it on my Mac, I deleted all gems for the ruby 2.1.0p0 version and used gem install dropcaster --pre.  I get the following loading errors:
- Bundler
- active_support/core_ext/logger

After installing these with gem, the gem install dropcaster --pre does the trick on my Ubuntu and Mac. The last time I tested it was about a month ago. I still had the error back then. So, I suppose that some of the recent commits fixed it.  I still get the active_support error if I do not use the pre-flag on all machines.

From my point of view, the ticket can be closed, but it feels strange for me to not have really figured out the cause.
